src/bin/pg_autoctl/pgctl.c

```diff
@@ -2046,6 +1917,69 @@ prepare_conninfo_sslmode(PQExpBuffer buffer, SSLOptions sslOptions)
 }
 
 
+/*
+ * pgctl_identify_system connects with replication=1 to our target node and run
+ * the IDENTIFY_SYSTEM command to check that HBA is ready.
+ */
+bool
+pgctl_identify_system(ReplicationSource *replicationSource)
+{
+	NodeAddress *primaryNode = &(replicationSource->primaryNode);
+
+	char primaryConnInfo[MAXCONNINFO] = { 0 };
+	char primaryConnInfoReplication[MAXCONNINFO] = { 0 };
+	PGSQL replicationClient = { 0 };
+	int len = 0;
+
+	/* sometimes we don't have a primary, just pretend everything is fine */
+	if (IS_EMPTY_STRING_BUFFER(primaryNode->host))
```

Comment:
I'd  rather  move this check to where `pgctl_identify_system` is called as  we used to have:
```C
!IS_EMPTY_STRING_BUFFER(primaryNode->host) &&	
			!pg_receivewal(pgdata, pg_ctl, replicationSource, "0/1", LOG_DEBUG)
```

Otherwise, this check/comment seems hard to follow and unexpected

Reply:
Good point, it's easier to read that way. Done now.